### DISCUSSION TITLE
player_activities

### DISCUSSION THREAD

**Discussion by LohTek** [CleverRaven/Cataclysm-DDA#46983] [Ideas]
The current activity system seems to be deeply flawed, requiring 8000 kcals to simply sew clothing is a bit excessive in my opinion. As I was digging a hole in the ground (in the game), I thought about the actions I was taking; swinging my arms to dig, my body shouldn't be consuming kcals immediately when I am doing this however I am seriously physically exerting myself and my stamina should be going down. 

My suggestion is to change the system to physical exertion consumes stamina; and kcals are used to replenish stamina. It appears to me it would be a better setup because when you fight and run you use stamina, when you dig and saw metal you are using the same muscle groups though in less intensity. Digging an entire trench would be really physically demanding on anyone and would require frequent breaks, that is where the stamina loss comes in.

Players would be required to rest after heavy activity and consume food/water to replenish their stamina. It would make the system more realistic because results are delayed instead of immediately losing kcals from crafting you lose it over time while walking, running, fighting, crafting, digging etc.

Feedback appreciated.

**Comment by anothersimulacrum** [comment]
> The current activity system seems to be deeply flawed, requiring 8000 kcals to simply sew clothing is a bit excessive in my opinion.

I strongly agree, and this is not what the current system reproduces.

The issue for weariness talks about stamina, and there could (and probably should), be more intersection between these two systems.
#42343

Stamina as it currently is, however, is more of an aerobic exhaustion measure, and weariness is an energy-based one, to my admittedly limited understanding.

**Reply by LohTek** [reply]
> Stamina as it currently is, however, is more of an aerobic exhaustion measure

That is my understanding of it as well; and I thought about physical exertion in those terms. Here's an example from the code the action
`ACT_FURNITURE_MOVE` 
So what we are doing here is dragging a piece of furniture across the floor. If you have ever moved a sofa, or any heavy furniture you would know this is very aerobic, after awhile of dragging this thing you become winded and need to rest. Same could be said for dragging objects along the ground, moving a large pile of stuff requires walking back and forth bending, lifting, and walking. Swinging a hammer, cutting wood, carrying materials etc; all very aerobic activities. Stamina loss should occur, and not an immediate cost of kcals; but a cost to replenish the stamina loss in terms of kcals. In real life after working out for an hour at the gym you become winded, your body begins to metabolize kcals and you then become hungry. At least that is my experience.
